Tests/LegacyUnit/FrontEnd/DefaultControllerTest.php

```diff
@@ -9452,6 +9453,161 @@ public function myEventsListForModifyMyEventsListRowHookWithoutInterfaceThrowsEx
+            'tx_seminars_seminars',
+            [
+                'pid' => $this->systemFolderPid,
+                'object_type' => \Tx_Seminars_Model_Event::TYPE_TOPIC,
+            ]
+        );
+        $this->testingFramework->createRecord(
+            'tx_seminars_seminars',
+            [
+                'pid' => $this->systemFolderPid,
+                'object_type' => \Tx_Seminars_Model_Event::TYPE_DATE,
+                'topic' => $topic,
+                'begin_date' => $GLOBALS['SIM_EXEC_TIME'] + 1000,
+                'end_date' => $GLOBALS['SIM_EXEC_TIME'] + 2000,
+            ]
+        );
+        $this->subject->setConfigurationValue('what_to_display', 'topic_list');
+
+        $hook = $this->createMock(SeminarListView::class);
+        $hook->expects(self::once())->method('modifyListHeader')->with($this->subject);
+        $hook->expects(self::once())->method('modifyListRow')->with($this->subject);
+        $hook->expects(self::never())->method('modifyMyEventsListRow');
+        $hook->expects(self::once())->method('modifyListFooter')->with($this->subject);
+        $hook->expects(self::once())->method('modifyEventBagBuilder')->with($this->subject, self::anything(), 'topic_list');
+        // We don't test for the second parameter (the bag builder instance here)
+        // because we cannot access it from the outside.
+        $hook->expects(self::never())->method('modifyRegistrationBagBuilder');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][SeminarListView::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
+
+        $this->subject->main('', []);
+    }
+
+    /**
+     * @test
+     */
+    public function listViewCallsSeminarListViewHookMethodsForSeminarList()
+    {
+        $this->subject->setConfigurationValue('what_to_display', 'seminar_list');
+
+        $hook = $this->createMock(SeminarListView::class);
+        $hook->expects(self::once())->method('modifyListHeader')->with($this->subject);
+        $hook->expects(self::once())->method('modifyListRow')->with($this->subject);
+        $hook->expects(self::never())->method('modifyMyEventsListRow');
+        $hook->expects(self::once())->method('modifyListFooter')->with($this->subject);
+        $hook->expects(self::once())->method('modifyEventBagBuilder')->with($this->subject, self::anything(), 'seminar_list');
+        // We don't test for the second parameter (the bag builder instance here)
+        // because we cannot access it from the outside.
+        $hook->expects(self::never())->method('modifyRegistrationBagBuilder');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][SeminarListView::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
+
+        $this->subject->main('', []);
+    }
+
+    /**
+     * @ test "events next day", "other dates" ?
```

Comment:
Yes, I think we should have those.

Reply:
Please advice on how to trigger them from the tests. They are part of the single view, if I remember correctly. But I have never used them myself.

Reply:
One of them should suffice to test.

For "other dates", you'll need to have a topic record (preferably in the future) with two date records referencing the topic record. A single view for one of the dates will list the other date for the same topic.

Reply:
OK, this raises another problem: The single view triggers both, "events_next_day" and "other_dates" in row. Always, at least the BagBuilder is built. Is it possible to do something like:

```php
$hook->expects(self::first())->method('modifyEventBagBuilder')->with($this->subject, self::anything(), 'events_next_day');
$hook->expects(self::second())->method('modifyEventBagBuilder')->with($this->subject, self::anything(), 'other_dates');
```

Reply:
Yes, with `->expects(self::at(0))->method` etc.